src/cluster.c

```diff
@@ -6020,7 +6020,8 @@ int clusterRedirectBlockedClientIfNeeded(client *c) {
     if (c->flags & CLIENT_BLOCKED &&
         (c->btype == BLOCKED_LIST ||
          c->btype == BLOCKED_ZSET ||
-         c->btype == BLOCKED_STREAM))
+         c->btype == BLOCKED_STREAM||
+         c->btype == BLOCKED_MODULE))
```

Comment:
i think we may need to check `moduleClientIsBlockedOnKeys` since a module can be blocked waiting for a module background thread, not a key creation.
also, not sure if we don't need a slightly more complicated code for releasing it.

Reply:
1. yes, I guess we need `moduleClientIsBlockedOnKeys`
2. `unblockClient` can handle modules that are blocked on keys sh it should be fine

Reply:
Only when we use  RedisModule_BlockClientOnKeys, these codes can be involved. So  we may not  need moduleClientIsBlockedOnKeys here.
Am I right?

Reply:
Only when we use `RedisModule_BlockClientOnKeys`, these codes can be involved. So we may not need`moduleClientIsBlockedOnKeys` here.
Am I right?

Reply:
I'll add that this function is weird in that the caller is the one that executes unblockClient if this function returns 1. It's a bit ugly. +1 to the test.

Reply:
I've tested it in my environment. Using `tests/modules/blockonkeys.c` may be enough. The critical is that we must migrate the hash slot which the key belongs to when the client is blocking on  this key.
Am I right?

Reply:
@qetu3790 AFAIK the current tests in `tests/modules/blockonkeys.c` don't cover this code (cluster isn't enabled), so i don't understand what you mean by "may be enough"?
if we wanna cover this code we need to add a test, either in the cluster tests (`tests/cluster/tests`), or fake cluster mode in `tests/unit/moduleapi/`, can you try to handle it?

> Only when we use RedisModule_BlockClientOnKeys, these codes can be involved

i don't understand, are you saying that this code will be unreachable when a module used RM_BlockClient?

Reply:
OK. I will add the tests ASAP.

Reply:
[The test is added.](https://github.com/redis/redis/pull/9483)

Reply:
I have added the test here  #9483, please check.